src/Core/Services/Implementations/CipherService.cs

```diff
@@ -790,5 +795,35 @@ private async Task<bool> UserCanEditAsync(Cipher cipher, Guid userId)
 
             return await _cipherRepository.GetCanEditByIdAsync(userId, cipher.Id);
         }
+
+        private void ValidateCipherLastKnownRevisionDateAsync(Cipher cipher, DateTime? lastKnownRevisionDate)
+        {
+            if (cipher.Id == default || !lastKnownRevisionDate.HasValue)
+            {
+                return;
+            }
+
+            if (cipher.RevisionDate - lastKnownRevisionDate > TimeSpan.FromSeconds(1))
+            {
+                throw new BadRequestException(
+                    "The cipher you are updating is out of date. Please save your work, sync your vault, and try again."
+                );
+            }
+        }
+
+        private void ValidateMultipleCipherLastKnownRevisionDatesAsync(
+            IEnumerable<(Cipher cipher, DateTime? lastKnownRevisionDate)> ciphers)
+        {
+            var outOfDateCiphers = ciphers.Where(c => c.cipher.Id != default &&
+                c.lastKnownRevisionDate != null &&
+                c.cipher.RevisionDate - c.lastKnownRevisionDate > TimeSpan.FromSeconds(1));
+
+            if (outOfDateCiphers.Any())
+            {
+                throw new BadRequestException(
+                    "At least one cipher you are updating is out of date. Please save your work, sync your vault, and try again."
+                );
+            }
```

Comment:
Would it be better to just loop and call your individual method above? `ValidateCipherLastKnownRevisionDateAsync`. That way you dont need to do the comparison logic twice

Reply:
Good point, I implemented this when still reading from the database, so it made sense to group. Without that your method is definitely better.